src/Utils.cc

```diff
@@ -563,4 +574,271 @@ thread_type_t fbgemmGetThreadPartition(
   return th_info;
 }
 
+namespace {
+
+// implementation taken from pytorch/c10/util/llvmMathExtras.h
```

Comment:
I'm worried about the code duplication here. @q10 - I don't think Torch is used as a dependency for `fbgemm`, but it is used as a dependency for `fbgemm_gpu`. If code duplication is common, perhaps we should make torch a dependency for all of `fbgemm` and use torch's libraries?

Reply:
That is something we will need to discuss with the fbgemm folks, since such a change will affect downstream users that use fbgemm without torch.

Reply:
I don't have enough knowledge about PyTorch and FBGEMM codebase to make a valuable proposal, but if we decide to do this, I suggest covering this in a separate PR (of course I could do this by myself).